contracts/KyberProxy.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity ^0.4.24;
+
+import "./vendors/kyber/KyberNetworkProxy.sol";
+import "./vendors/kyber/ERC20Interface.sol";
+import "./interfaces/TokenConverter.sol";
+import "./utils/Ownable.sol";
+
+contract KyberProxy is TokenConverter, Ownable {
+    
+    uint256 private constant MAX_UINT = uint256(0) - 1;
+
+    KyberNetworkProxy public converter;
+    Token ethToken;
+
+    event Swap(address indexed sender, ERC20 srcToken, ERC20 destToken, uint amount);
+
+    constructor(Token _ethToken) public {
+        ethToken = _ethToken;
+    }
+
+    function getReturn(
+        Token srcToken, 
+        Token destToken, 
+        uint256 srcQty
+    ) external view returns (uint256) {
+        (uint256 amount,) = converter.getExpectedRate(ERC20(srcToken), ERC20(destToken), srcQty);
+        return amount;
+    }
+
+    function convert(
+        Token srcToken,
+        Token destToken, 
+        uint256 srcQty, 
+        uint256 minReturn
+    ) external payable returns (uint256 amount) {
+
+        require(msg.value == 0, "ETH not required");
```

Comment:
If convert is payable it should be able to receive ETH, a Token converter should work ERC20 <-> ERC20 and ETH <-> ERC20